src-v2/NTS.Persistance/Adapters/CompetitionRepository.cs

```diff
@@ -1,15 +1,34 @@
 ﻿using NTS.Domain.Setup.Entities;
 using Not.Exceptions;
+using Not.Application.Ports.CRUD;
 
 namespace NTS.Persistence.Adapters;
 
-public class CompetitionRepository : RepositoryBase<Competition, State>
+public class CompetitionRepository : ParentRepository<Competition, Event, State>, IParentRepository<Competition>
```

Comment:
Close, but not quite there yet. What is the downside of having separate implementation as `IParetnRepository<Competition>`?